src/Antd_DatePicker.re

```diff
@@ -143,7 +143,7 @@ module RangePicker = {
         ~showTime: 'f=?, /* TimePickerProps | boolean; */
         ~ranges: 'g=?, /* todo: {[range: string]: RangePickerPresetRange,}; */
         ~placeholder: (string, string)=?,
-        ~mode: array(string)=?, /* todo: how do i do ? string | string[]; */
+        ~mode: string=?, /* todo: how do i do ? string | string[]; */
```

Comment:
You can make a mode GADT and modeToProps like tou’re doing with othe props

Reply:
it is using rc-calendar internally and just passing the mode to it. 
enum('time', 'date', 'month', 'year', 'decade') is what those docs say?

Should we stick with the typescript from ant-design, or should we use a polymorphic variant.

not sure why they have it set to string | string[],  they don't have mode listed in their docs.

Reply:
Hmm. I would say convert them to poly variants. String enums are the source of bugs